test/jdk/sun/security/pkcs11/Provider/MultipleLogins.java

```diff
@@ -30,35 +41,38 @@
 import javax.security.auth.callback.PasswordCallback;
 import javax.security.auth.callback.UnsupportedCallbackException;
 import javax.security.auth.login.LoginException;
+
 import java.io.IOException;
 import java.lang.ref.WeakReference;
 import java.security.*;
 
 import jdk.test.lib.util.ForceGC;
 import jtreg.SkippedException;
 
-public class MultipleLogins {
-    private static final String KS_TYPE = "PKCS11";
+public class MultipleLogins extends PKCS11Test {
     private static final int NUM_PROVIDERS = 20;
     private static final SunPKCS11[] providers = new SunPKCS11[NUM_PROVIDERS];
 
     public static void main(String[] args) throws Exception {
-        String nssConfig = null;
-        try {
-            nssConfig = PKCS11Test.getNssConfig();
-        } catch (SkippedException exc) {
-            System.out.println("Skipping test: " + exc.getMessage());
-        }
+        // This bypasses the PKCS11Test settings and run the mandatory
+        // main method directly. This is needed to keep the custom logic of the test
+        new MultipleLogins().main((Provider)null);
+    }
+
+    @Override
+    public void main(Provider p) throws Exception {
+        copyNssCertKeyToClassesDir();
+
+        String nssConfig = getNssConfig();
 
         if (nssConfig == null) {
             // No test framework support yet. Ignore
-            System.out.println("No NSS config found. Skipping.");
-            return;
+            throw new SkippedException("No NSS config found. Skipping.");
         }
 
-        for (int i =0; i < NUM_PROVIDERS; i++) {
+        for (int i = 0; i < NUM_PROVIDERS; i++) {
             // loop to set up test without security manger
-            providers[i] = (SunPKCS11)PKCS11Test.newPKCS11Provider();
+            providers[i] = (SunPKCS11)newPKCS11Provider();
```

Comment:
Not sure if it's worth updating, but since `configure` always return a new provider, there is no need to call `newPKCS11Provider` here and we can just call `configure` on the same `Security.getProvider("SunPKCS11")`. Also, you can use the public `AuthProvider` instead of internal class `SunPKCS11` everywhere.